opentracing_shim/src/main/java/io/opentelemetry/opentracingshim/Propagation.java

```diff
@@ -48,7 +49,9 @@ public SpanContextShim extractTextFormat(TextMapExtract carrier) {
         tracer().getHttpTextFormat().extract(carrierMap, TextMapGetter.INSTANCE);
     io.opentelemetry.correlationcontext.CorrelationContext distContext =
         contextManager().getHttpTextFormat().extract(carrierMap, TextMapGetter.INSTANCE);
-
+    if (context == SpanContext.getInvalid()) {
+      throw new IllegalArgumentException("Extracted span context is invalid");
```

Comment:
Is this the right answer here, or are we pushing the problem one level up the stack?

Reply:
In OT we are **not** throwing any exceptions  at the propagation layer, merely returning null for `extract()`. I suggest this part remains untouched, and another PR handles this properly (either you or me can write that one ;) )

Reply:
I wanted to keep behavior for shim, otherwise  it starts to return not null but INVALID span context
```
  @Test
  public void extract_nullContext() {
    SpanContext result =
        tracerShim.extract(
            Format.Builtin.TEXT_MAP, new TextMapAdapter(Collections.<String, String>emptyMap()));
    assertNull(result);
  }
``` 
as i get u right - should i change this test assert to Span.Invalid?

Reply:
Right, I had forgotten about this test case. So lets keep this check - and in general I'd be up for actually checking if the `SpanContext` is invalid (which also is mentioned in the Specification):

```java
if (!spanContext.isValid()) {
  return null;
}
```